src/lib/parallel_scan/parallel_scan.ml

```diff
@@ -304,10 +304,14 @@ module State = struct
     | true, Merge (Bcomp _), Merged z ->
         let%bind () =
           if cur_pos = 0 then (
-            t.acc <- (fst t.acc |> Int.( + ) 1, Some z) ;
-            t.other_trees_data
-            <- List.take t.other_trees_data (List.length t.other_trees_data - 1) ;
-            Ok () )
+            match List.rev t.other_trees_data with
+            | [] ->
+                failwith
+                  "Not possible, if we're emitting something, we have data"
+            | last :: rest ->
+                t.acc <- (fst t.acc |> Int.( + ) 1, Some (z, last)) ;
+                t.other_trees_data <- List.rev rest ;
+                Ok () )
```

Comment:
`last` here needs to be reversed as well. New transactions are added at the front.